src/ponyjs/models/base.js

```diff
@@ -56,14 +53,12 @@ class ModelState {
 
 
 class ModelBase {
-  constructor(data={}) {
+  constructor(data) {
 
     // set up the instance state
     this._state = data;
 
-    for (let key of Object.keys(data)) {
-      this[key] = data[key]
-    }
+    Object.assign(this, data);
```

Comment:
Object.assign doesn't throw an error when source is undefined, so it all works, and yes there're 3 test cases where it is undefined so it's covered.